src/ActivityRepository.js

```diff
@@ -85,31 +85,33 @@ class ActivityRepository {
     return this.user.slice(index - 6, index + 1);
   }
 
-  getPositiveStepTrends() {
-    return this.user.reduce((acc, day, index) => {
-      if (index < 2) {
+  getStepTrends(type) {
+    if (type === 'positive') {
+      return this.user.reduce((acc, day, index) => {
+        if (index < 2) {
+          return acc;
+        }
+        if ((day.numSteps > this.user[index - 1].numSteps) &&
+          (this.user[index - 1].numSteps > this.user[index - 2].numSteps)) {
+          acc.push(day.date);
+        }
         return acc;
-      }
-      if ((day.numSteps > this.user[index - 1].numSteps) &&
-        (this.user[index - 1].numSteps > this.user[index - 2].numSteps)) {
-        acc.push(day.date);
-      }
-      return acc;
-    }, []);
-  }
-
-  getNegativeStepTrends() {
-    return this.user.reduce((acc, day, index) => {
-      if (index < 2) {
+      }, []);
+    } else {
+      return this.user.reduce((acc, day, index) => {
+        if (index < 2) {
+          return acc;
+        }
+        if ((day.numSteps < this.user[index - 1].numSteps) &&
+          (this.user[index - 1].numSteps < this.user[index - 2].numSteps)) {
+          acc.push(day.date);
+        }
         return acc;
+        }, []);
       }
-      if ((day.numSteps < this.user[index - 1].numSteps) &&
-        (this.user[index - 1].numSteps < this.user[index - 2].numSteps)) {
-        acc.push(day.date);
-      }
-      return acc;
-    }, []);
+    }
```

Comment:
As we kind of saw yesterday, there's not a great way to DRY up this repetitive code as it only differs based on the greater than/less than operator. The only solution I could think of, which is likely even HAIRIER is doing something like:

```js
if type === positive
    comparisonOperator = '>';
else 
    comparisonOperator = '<'
```

then in your conditional logic you would swap out the greater than / less than signs for `eval(comparisonOperator)`

This would let you use a variable to replace the hardcoded greater than / less than signs. But you need to use this weird `eval` function that tells JavaScript "treat the string thats passed in as some sort of JavaScript that actually needs to be evaluated"

Long story short, dont do this. Just leave that code as-is for now despite its ugliness,

Reply:
@trishalanglois @allisonjw Per Brittany's comment, do you think we should close this PR and shelf the refactoring on this for now?